xarray/tests/test_concat.py

```diff
@@ -41,9 +41,6 @@ def test_concat_compat():
     for var in ["has_x", "no_x_y"]:
         assert "y" not in result[var]
 
-    with raises_regex(ValueError, "'q' is not present in all datasets"):
```

Comment:
This no longer raises a nice error message. I'm not sure what to do in this case: one dataset has a labeled coordinate "q" and the other doesn't have "q" at all. Currently this will error with "ValueError: One or more of the specified variables cannot be found in this dataset" which is fine ( I think)

Reply:
Agree—naming it is ideal but that's fine if we can fix it quickly

Reply:
Can we keep the test which asserts it raises an error though?

Reply:
Done.